.eslintrc.cjs

```diff
@@ -93,6 +93,7 @@ const tsRules = defineConfig({
 		// note you must disable the base rule as it can report incorrect errors
 		"no-shadow": "off",
 		"@typescript-eslint/no-shadow": ["error"],
+		"@typescript-eslint/no-redundant-type-constituents": "off",
```

Comment:
don't really see the point in making our types worse, even though it's technically correct that just having `undefined` as a type encompasses everything, but documentation wise that's poor.